test/e2e/framework.go

```diff
@@ -1871,19 +1871,15 @@ func validatePodIP(podNetworkCIDR string, ip net.IP) (bool, error) {
 
 // CreateService creates a service with port and targetPort.
 func (data *TestData) CreateService(serviceName, namespace string, port, targetPort int32, selector map[string]string, affinity, nodeLocalExternal bool,
-	serviceType corev1.ServiceType, ipFamily *corev1.IPFamily) (*corev1.Service, error) {
+	serviceType corev1.ServiceType, ipFamilies []corev1.IPFamily) (*corev1.Service, error) {
 	annotation := make(map[string]string)
-	return data.CreateServiceWithAnnotations(serviceName, namespace, port, targetPort, corev1.ProtocolTCP, selector, affinity, nodeLocalExternal, serviceType, ipFamily, annotation)
+	return data.CreateServiceWithAnnotations(serviceName, namespace, port, targetPort, corev1.ProtocolTCP, selector, affinity, nodeLocalExternal, serviceType, ipFamilies, annotation)
 }
 
 // CreateServiceWithAnnotations creates a service with Annotation
 func (data *TestData) CreateServiceWithAnnotations(serviceName, namespace string, port, targetPort int32, protocol corev1.Protocol, selector map[string]string, affinity, nodeLocalExternal bool,
-	serviceType corev1.ServiceType, ipFamily *corev1.IPFamily, annotations map[string]string) (*corev1.Service, error) {
+	serviceType corev1.ServiceType, ipFamilies []corev1.IPFamily, annotations map[string]string) (*corev1.Service, error) {
```

Comment:
I think having 2 entries in `IPFamilies` without also configuring `IPFamilyPolicy` means that the Service will never be dual-stack (or am I wrong?). I am not sure this is what we want here. I thought we wanted the ability to have a dual-stack Service?

Reply:
You are right. We need to configure `IPFamilyPolicy ` as well. Otherwise the default value for `IPFamilyPolicy` is single-stack. I think we need to also configure the kind cluster to set `IP_FAMILY`=dual.
I'll update it later.